xyz-hub-service/src/main/java/com/here/xyz/httpconnector/config/SecretManagerCredentialsProvider.java

```diff
@@ -0,0 +1,65 @@
+package com.here.xyz.httpconnector.config;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.here.xyz.httpconnector.CService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class SecretManagerCredentialsProvider implements AWSCredentialsProvider  {
+
+    private static final Logger logger = LogManager.getLogger();
+    private static final int DEFAULT_REFRESH_INTERVAL_SECONDS = 3600;
+
+    private final AtomicReference<AWSCredentials> credentialsRef;
+    private final ScheduledExecutorService scheduler;
+
+    private String secretArn;
+
+    public SecretManagerCredentialsProvider(String secretArn) {
+        this(secretArn, DEFAULT_REFRESH_INTERVAL_SECONDS);
+    }
+
+    public SecretManagerCredentialsProvider(String secretArn, long refreshInterval) {
+
+        this.secretArn = secretArn;
+        this.credentialsRef = new AtomicReference<>();
+        this.scheduler = Executors.newScheduledThreadPool(1);
+
+        scheduleCredentialsRefresh(refreshInterval);
+    }
+
+    private void scheduleCredentialsRefresh(long refreshInterval) {
+        scheduler.scheduleAtFixedRate(this::refresh, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public AWSCredentials getCredentials() {
+
+        AWSCredentials currentCredentials = credentialsRef.get();
+
+        if(currentCredentials == null) {
+            refresh();
+            currentCredentials = credentialsRef.get();
+        }
+
+        return currentCredentials;
+
+    }
+
+    @Override
+    public void refresh() {
+        try {
+            AWSCredentials newCredentials = CService.jobSecretClient.getCredentialsFromSecret(secretArn);
+            credentialsRef.set(newCredentials);
+        } catch (Exception e) {
+            logger.error("Failed to refresh credentials from secret manager: {}", e.getMessage());
+            throw new RuntimeException("Failed to refresh credentials from secret manager", e);
```

Comment:
I think we should not (re-)throw an exception here, because that would surpress all subsequent refreshs. 

See: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleAtFixedRate-java.lang.Runnable-long-long-java.util.concurrent.TimeUnit-

In case of recoverable errors (e.g. service / connection related) the next refresh would have the chance to reload the credentials again.

Reply:
Resolved by switching to credentials provider solution.